assets/chat/js/messages/ChatBroadcastMessage.js

```diff
@@ -0,0 +1,65 @@
+import { usernameColorFlair } from './ChatUserMessage';
+import ChatEventMessage from './ChatEventMessage';
+import MessageTypes from './MessageTypes';
+
+export default class ChatBroadcastMessage extends ChatEventMessage {
+  constructor(message, user, timestamp = null) {
+    super(message, timestamp);
+    this.type = MessageTypes.BROADCAST;
+    this.user = user;
+  }
+
+  buildUserTemplate(chat = null) {
+    // User with an id of -1 is the dummy system user used for server broadcasts
+    if (this.user.id !== -1) {
```

Comment:
Let's make this self-documenting by instead defining a method on `ChatUser` that returns true if the user is the system user. I'd also prefer a guard clause that returns early for the system user to make this method a little easier to read.

Reply:
Done!